# HG changeset patch
# User Torsten <mttl@mailbox.org>
# Date 1503692974 -7200
#      Fri Aug 25 22:29:34 2017 +0200
# Node ID cf16f65211803b850c2c59e615f63fa0026f457d
# Parent  0bc54dbc2bfcf15cffb5cd3b22b27612cd53e972
double clicking a variable in workspace view opens variable editor (bug #51843)

* workspace-view.cc (workspace_view): connect signal for double clicking
  in the table view with the already existing slot for opening the editor;
  (contextmenu_requested): put open variable editor at first position of the
  context menu

diff --git a/libgui/src/workspace-view.cc b/libgui/src/workspace-view.cc
--- a/libgui/src/workspace-view.cc
+++ b/libgui/src/workspace-view.cc
@@ -145,16 +145,19 @@ workspace_view::workspace_view (QWidget 
   connect (_filter_checkbox, SIGNAL (toggled (bool)),
            this, SLOT (filter_activate (bool)));
   connect (_filter->lineEdit (), SIGNAL (editingFinished ()),
            this, SLOT (update_filter_history ()));
 
   connect (view, SIGNAL (customContextMenuRequested (const QPoint&)),
            this, SLOT (contextmenu_requested (const QPoint&)));
 
+  connect (view, SIGNAL (doubleClicked (QModelIndex)),
+           this, SLOT (handle_contextmenu_edit (void)));
+
   connect (this, SIGNAL (command_requested (const QString&)),
            p, SLOT (execute_command_in_terminal (const QString&)));
 
   connect (this, SIGNAL (edit_variable_signal (const QString&)),
            p, SLOT (edit_variable (const QString&)));
 }
 
 void
@@ -304,16 +307,19 @@ workspace_view::contextmenu_requested (c
 
   QModelIndex index = view->indexAt (qpos);
 
   // if it isnt Local, Glocal etc, allow the ctx menu
   if (index.isValid () && index.column () == 0)
     {
       QString var_name = get_var_name (index);
 
+      menu.addAction (tr ("Open in Variable Editor"), this,
+                      SLOT (handle_contextmenu_edit ()));
+
       menu.addAction (tr ("Copy name"), this,
                       SLOT (handle_contextmenu_copy ()));
 
       menu.addAction (tr ("Copy value"), this,
                       SLOT (handle_contextmenu_copy_value ()));
 
       QAction *rename = menu.addAction (tr ("Rename"), this,
                                         SLOT (handle_contextmenu_rename ()));
@@ -322,19 +328,16 @@ workspace_view::contextmenu_requested (c
       const workspace_model *wm = static_cast<const workspace_model *> (m);
 
       if (! wm->is_top_level ())
         {
           rename->setDisabled (true);
           rename->setToolTip (tr ("Only top-level symbols may be renamed"));
         }
 
-      menu.addAction (tr ("Open in Variable Editor"), this,
-                      SLOT (handle_contextmenu_edit ()));
-
       menu.addSeparator ();
 
       menu.addAction ("disp (" + var_name + ')', this,
                       SLOT (handle_contextmenu_disp ()));
 
       menu.addAction ("plot (" + var_name + ')', this,
                       SLOT (handle_contextmenu_plot ()));
 
